File: esp32/Linux/JavaFX/src/main/kotlin/com/example/trenta4/MyClient.kt
package com.example.trenta4

import io.ktor.client.HttpClient
import io.ktor.client.engine.cio.CIO
import io.ktor.client.request.get
import io.ktor.client.request.header
import io.ktor.client.statement.HttpResponse
import io.ktor.client.statement.bodyAsText
import io.ktor.utils.io.errors.IOException

object MyClient {

    private val client = HttpClient(CIO)
    private const val url = "https://2desperados.it/esp32/searchLastNCoords/1"
    private val apiKey: String = System.getenv("ESP32_API_KEY")

    suspend fun read(): String {

        return try {
            val response: HttpResponse = client.get(url) {
                header("X-Api-Key", apiKey)
            }

            response.bodyAsText()

        } catch (e: IOException) {

            println(e.stackTrace)
            "IOException"
        }
    }
}

File: esp32/Web/src/MyClient.ts
interface MyClient { fetch: () => Promise<string> }

export const myClient: MyClient = {
    
    fetch: async function() {

        const url = 'https://2desperados.it/esp32/searchLastNCoords/1'

        try {
            
            const response = await fetch(url, {
              
                headers: { 'X-api-key': '' }
            })
            
            return await response.text()
        
        } catch (error) { console.error(error) }
    }
}
